src/bms/parse.rs

```diff
@@ -1,27 +1,26 @@
-//! Parser for BMS format. The reason why the implementation separated into lex and parse is the score may contain some randomized elements such as `#RANDOM`. This separation make us able to parse the tokens with the custom random generator cheaply.
+//! BMS: Parse Module.
+//!
+//! Lex => [Parse]
```

Comment:
This doc comment seems to too lack of descriptions. There should be a guide intended for developers to read.